torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -236,6 +236,29 @@ std::pair<at::Tensor, std::shared_ptr<torch::lazy::Value>> ReduceScatter(
       std::make_shared<torch::lazy::Value>(new_token));
 }
 
+std::pair<std::vector<at::Tensor>, std::shared_ptr<torch::lazy::Value>>
+ReduceScatterCoalesced(const std::string& reduce_type,
+                       const std::vector<at::Tensor>& outputs,
+                       const std::vector<at::Tensor>& inputs,
+                       const std::shared_ptr<torch::lazy::Value>& token,
+                       double scale, int64_t scatter_dim, int64_t shard_count,
+                       const std::vector<std::vector<int64_t>>& replica_groups,
+                       bool pin_layout) {
+  std::vector<XLATensorPtr> xtensors_out =
+      GetXlaTensors(outputs, /*want_all=*/true);
```

Comment:
if you want to handle the `outputs` cases, it is better to define a `ReduceScatterCoalescedOut` instead of merging the logic into a single function.

Reply:
Is it ok if I work on this in another change?

Reply:
we can work on the `reduce_scatter_out` in a separate pr

Reply:
Thanks Jack. Here's the change to error out if output!=None: https://github.com/pytorch/xla/pull/5956/commits/9baadf50994a77cc578f5166304484a4c98ab0ac

Reply:
https://github.com/pytorch/xla/pull/6058 is for adding out-of-place version for reduce-scatter, along with https://github.com/pytorch/xla/pull/6059 for out-of-place all-gather.